test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -524,6 +560,163 @@ protected static IndexReader maybeWrapReaderEs(DirectoryReader reader) throws IO
+    }
+
+    public void testSupportedFieldTypes() throws IOException {
+
+        MapperRegistry mapperRegistry = new IndicesModule(Collections.emptyList()).getMapperRegistry();
+        Settings settings = Settings.builder().put("index.version.created", Version.CURRENT.id).build();
+        String fieldName = "typeTestFieldName";
+        List<ValuesSourceType> supportedVSTypes = getSupportedValuesSourceTypes();
+
+        if (supportedVSTypes.isEmpty()) {
+            // If the test says it doesn't support any VStypes, it has not been converted yet so skip
+            return;
+        } else if (supportedVSTypes.contains(CoreValuesSourceType.ANY)) {
+            throw new IllegalArgumentException("Tests should not specify CoreValuesSourceType.ANY as a supported ValuesSourceType, " +
+                "but should instead list the concrete ValuesSourceTypes that are supported");
+        }
+
+        for (Map.Entry<String, Mapper.TypeParser> mappedType : mapperRegistry.getMapperParsers().entrySet()) {
+            if (TYPE_TEST_BLACKLIST.contains(mappedType.getKey())) {
+                continue;
+            }
+
+            if (mappedType.getKey().equals(ObjectMapper.CONTENT_TYPE)) {
+                // Cannot aggregate objects
+                continue;
+            }
+
+            Map<String, Object> source = new HashMap<>();
+            source.put("type", mappedType.getKey());
+            source.put("doc_values", "true");
+
+            FieldMapper mapper = null;
+            try {
+                Mapper.Builder builder = mappedType.getValue().parse(fieldName, source, new MockParserContext());
+                mapper = (FieldMapper) builder.build(new BuilderContext(settings, new ContentPath()));
+            } catch (Exception e) {
+                fail();
+            }
+            MappedFieldType fieldType = mapper.fieldType();
+
+
+            try (Directory directory = newDirectory()) {
+                RandomIndexWriter indexWriter = new RandomIndexWriter(random(), directory);
+                writeTestDoc(fieldType, fieldName, indexWriter);
+                indexWriter.close();
+
+                try (IndexReader indexReader = DirectoryReader.open(directory)) {
+                    IndexSearcher indexSearcher = newSearcher(indexReader, true, true);
+                    AggregationBuilder aggregationBuilder = createAggBuilderForTypeTest(fieldType, fieldName);
+
+                    try {
+                        searchAndReduce(indexSearcher, new MatchAllDocsQuery(), aggregationBuilder, fieldType);
+                        if (supportedVSTypes.contains(fieldType.getValuesSourceType()) == false) {
+                            fail("Aggregator [" + aggregationBuilder.getType() + "] should not support field type ["
+                                + fieldType.typeName() + "] but executing against the field did not throw an excetion");
+                        }
+                    } catch (Exception e) {
```

Comment:
Once we are running these tests against aggregations using the VS Registry, we can say reliably what exception should be thrown.  At that point, we should invert this test - if the VSType is not supported, we should use `expectThrows` to confirm we get the right exception, and otherwise fail on any exception.  Probably can't do that now though since our exceptions are a bit inconsistent in master.